### PR TITLE
Automated SSL certificate generation and management

### PR DESCRIPTION
**Pull request by LJ-Software**
# Description

Update documentation and the offline installation script to support cert-manager being added to the karavi-observability helm chart.

# Issues

List the issues impacted by this PR:

| Issue ID |
| -------- |
|    #1       |

# Checklist:

- [x] I have performed a self-review of my own changes.

### CODE REVIEW

**Comment by ioeicicdsvc** [comment]
@LJ-Software: Thank you for submitting your pull request!

Please make sure you have followed our contributing guidelines. We will review it as soon as possible.

------------------------------------------------------------------------------------------------------------------------
We want your feedback! If you have any questions or suggestions regarding our contributing process/workflow, please reach out to us at karavi@dell.com.

File: docs/GETTING_STARTED_GUIDE.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ You may obtain a copy of the License at
...
+This project provides Kubernetes administrators insight into CSI (Container Storage Interface) Driver persistent storage topology, usage and performance. Metrics data is collected and pushed to the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector), so it can be processed, and exported in a format consumable by Prometheus. Topology data related to containerized volumes that are provisioned by a CSI (Container Storage Interface) Driver is also captured. The metrics and topology data are visualized through Grafana dashboards. SSL certificates for TLS between nodes are handled by [cert-manager](https://github.com/jetstack/cert-manager).
```
Comment (tdawe) [inline docs/GETTING_STARTED_GUIDE.md:13]: The link for cert-manager should open in a new tab. @medegw01 is currently adding this to other external links across the repositories.

Comment (tdawe) [reply docs/GETTING_STARTED_GUIDE.md:13]: It seems like opening in new tabs is not supported in GitHub Markdown, so ignore this review comment.

File: docs/GETTING_STARTED_GUIDE.md [hunk 2]
```diff
@@ -49,20 +49,31 @@ This project is deployed using Helm. The supported version of Helm is listed bel
 
 ## Installing the Chart
 
+Before installing the karavi-observability chart, you must first install the cert-manager CustomResourceDefinition resources with the command below.
+
+```console
+$ kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v1.1.0/cert-manager.crds.yaml
+```
+
+The command below deploys Karavi Observability on the Kubernetes cluster in the default configuration. The [configuration](##Configuration) section below lists all the parameters that can be configured during installation.
```
[the remark this answers is not in the document]
Comment (LJ-Software) [reply docs/GETTING_STARTED_GUIDE.md:68]: Added a section above this titled "TLS Encryption".
Do you think it's sufficient?

File: installer/README.md [hunk 3]
```diff
@@ -111,7 +111,12 @@ or
 [user@anothersystem /home/user/offline-karavi-observability-bundle]# cd helm
 ```
 
-2. Now that the required images have been made available and the Helm Charts configuration updated with references to the internal registry location, installation can proceed by following the instructions that are documented within the Helm charts repository.
+2. Install necessary cert-manager CustomResourceDefinitions provided:
+```
+[user@anothersystem /home/user/offline-karavi-observability-bundle/helm]# kubectl apply --validate=false -f cert-manager.crds.yaml
+```
+
+3. Now that the required images have been made available and the Helm Charts configuration updated with references to the internal registry location, installation can proceed by following the instructions that are documented within the Helm charts repository.
```
Comment (atye) [inline installer/README.md:119]: I believe "Helm Charts" should be possessive and you could add a "has been" before udpated.

...and the Helm Chart's configuration has been updated with....

Comment (LJ-Software) [reply installer/README.md:119]: Great catch. It looks like "Helm chart" is used in other places in the docs. I will change it accordingly.

**Comment by tdawe** [comment]
This PR should target the main branch and not the karavi-observability-0.2.0 branch. That release branch should be deleted since all changes are now going into main.

**Review by atye — DISMISSED** [review]

**Comment by LJ-Software** [comment]
> 
> 
> This PR should target the main branch and not the karavi-observability-0.2.0 branch. That release branch should be deleted since all changes are now going into main.

Updated. Thanks for pointing this out.

**Review by tdawe — APPROVED** [review]

**Review by atye — APPROVED** [review]

**Pull request merged**